management_api_app/db/repositories/workspaces.py

```diff
@@ -34,13 +35,20 @@ def get_workspace_by_workspace_id(self, workspace_id: UUID4) -> Resource:
         else:
             raise EntityDoesNotExist
 
-    def create_workspace(self, workspace_create: ResourceInCreate) -> Resource:
-        workspace = Resource(
+    def create_workspace(self, workspace_create: WorkspaceInCreate) -> Workspace:
+        workspace = Workspace(
             id=str(uuid.uuid4()),
-            resourceSpec=ResourceSpec(name=workspace_create.resourceSpec.name, version=workspace_create.resourceSpec.version),
-            parameters=workspace_create.parameters,
-            resourceType=ResourceType.Workspace,
-            status=Status.NotDeployed
+            friendlyName=workspace_create.friendlyName,
+            description=workspace_create.description,
+            resourceSpec=ResourceSpec(
```

Comment:
Hmm the actual spec should have come from the ResourceSpecs container using name and version. Is that done somewhere else or later?

Reply:
can you elaborate -- the workspaceType here is coming in from the user as the resourceSpec name (e.g. tre-vanilla-...), the version will be calculated as the latest for that spec in #167

Do you mean the parameters?

It becomes a little difficult as the parameters are both parameters that are fixed - that should be provided by the API
(the user can't give neither the workspace_id, location, core_id or really the address_space.) and potentially some that are not fixed... perhaps we should rethink this and make the ones that should be provided by the API move outside the params section.

Reply:
I am saying that when creating a workspace you receive name and version of the bundle. We should get the spec(which was pre registered with us) from the cosmos container ResourceSpecs using the given name and version. Then we know what spec to fill and ideally we would validate the presence of input parameters using this spec(but we are not doing that right now) Spec is something to read from the database.

Reply:
right... but it gets complicated, let's discuss in standup

Reply:
Resolving per offline discussions - moving this to a more flat structure to differentiate between ResourceSpec (the specification) and what we send as input (like in this case)